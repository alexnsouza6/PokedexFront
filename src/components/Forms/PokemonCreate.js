import React, { Component } from "react";
import { Button, Header } from "semantic-ui-react";
import { connect } from "react-redux";
import { createPokemon } from "./actions/PokemonActions";

export class PokemonCreate extends Component {
  state = {
    pokemonName: "",
    pokemonImage: {},
    pokemonType: "",
    pokemonEvolutionID: ""
  };

  handlePokemonNameChange = e => {
    this.setState({ pokemonName: e.target.value });
  };

  handlePokemonType = e => {
    this.setState({ pokemonType: e.target.value });
  };

  handlePokemonEvolution = e => {
    this.setState({ pokemonEvolutionID: parseInt(e.target.value) });
  };

  fileSelectedHandler = event => {
    const selectedFile = event.target.files;
    this.setState({ pokemonImage: selectedFile[0] });
  };

  handleSubmit = e => {
    e.preventDefault();
    const type = [
      { description: this.state.pokemonType },
      { description: "grass" }
    ];
    const fd = new FormData();
    fd.append("image_url", this.state.pokemonImage);
    fd.append("name", this.state.pokemonName);
    fd.append("evolution_id", this.state.pokemonEvolutionID);
    fd.append("pokemon_types[]", JSON.stringify(type));
    this.props.createPokemon(fd, () => this.props.history.push("/"));
  };

  render() {
    return (
      <div>
        <Header as="h1"> Create your own pokemon! </Header>
        <form onSubmit={this.handleSubmit}>
          <label htmlFor={this.state.pokemonName}> Pokemon Name </label>
          <input
            type="text"
            value={this.state.pokemonName}
            onChange={this.handlePokemonNameChange}
          />
          <br />
          <label htmlFor="pokemon-image"> Pokemon Picture </label>
          <input type="file" onChange={this.fileSelectedHandler} />
          <br />
          <label htmlFor={this.state.pokemonType}>Pokemon Type</label>
          <input
            type="text"
            value={this.state.pokemonType}
            onChange={this.handlePokemonType}
          />
          <br />
          <label htmlFor={this.state.pokemonEvolutionID}>
            Pokemon Evolution (Type its ID)
          </label>
          <input
            type="text"
            value={this.state.pokemonEvolutionID}
            onChange={this.handlePokemonEvolution}
          />
          <br />
          <Button type="submit"> Submit </Button>
        </form>
      </div>
    );
  }
}

export default connect(
  null,
  { createPokemon }
)(PokemonCreate);
