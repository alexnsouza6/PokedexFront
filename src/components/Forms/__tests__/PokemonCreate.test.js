import React from "react";
import renderer from "react-test-renderer";
import { PokemonCreate } from "../PokemonCreate";

it("renders correctly", () => {
  const tree = renderer.create(<PokemonCreate />).toJSON();
  expect(tree).toMatchSnapshot();
});
