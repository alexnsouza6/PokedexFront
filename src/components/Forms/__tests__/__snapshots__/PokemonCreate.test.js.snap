// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    className="ui header"
  >
     Create your own pokemon! 
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor=""
    >
       Pokemon Name 
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <label
      htmlFor="pokemon-image"
    >
       Pokemon Picture 
    </label>
    <input
      onChange={[Function]}
      type="file"
    />
    <br />
    <label
      htmlFor=""
    >
      Pokemon Type
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <label
      htmlFor=""
    >
      Pokemon Evolution (Type its ID)
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <button
      className="ui button"
      onClick={[Function]}
      type="submit"
    >
       Submit 
    </button>
  </form>
</div>
`;
